manifests/peer.pp

```diff
@@ -45,7 +45,7 @@
   $pool = 'default'
 ) {
 
-  $binary = $::gluster_binary
+  $binary = defined('$gluster_binary') ? { true => $::gluster_binary, false => false }
```

Comment:
@ross-w Thanks for your PR and the time you've taken to submit it.

Unfortunately using `defined()` for checking facts like this is broken on puppet versions before 3.7.5.
I've added a spec test and done a bit of playing around in https://github.com/voxpupuli/puppet-gluster/pull/85 to show this.

Hopefully I'll find time to get a final PR (one without my extra 'test' commits) submitted (and merged) tomorrow.